### PR TITLE
SAK-30712 More Sites - more flexibility for new site types.

### PR DESCRIPTION
**Pull request by danielmerino**

### CODE REVIEW

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java [hunk 1]
```diff
@@ -228,6 +228,24 @@ protected void processMySites()
 		// get Sections
 		Map<String, List> termsToSites = new HashMap<String, List>();
 		Map<String, List> tabsMoreTerms = new TreeMap<String, List>();
+		
+		//SAK-30712
+		boolean moresitesExternalConfig = Boolean.parseBoolean(serverConfigurationService.getString("moresites.externalConfig","false"));
```
Comment (ottenhoff) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java]: Please just use serverConfigurationService.getBoolean

Comment (danielmerino) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java]: Done, sorry.

**Comment by ottenhoff** [comment]
Okay LGTM

**Comment by juanjmerono** [comment]
I think you don't need 3 properties for this.
If someone define **moresites.externalConfig.siteTypes** is because he wants to activate this feature, so the **moresites.externalConfig=true** is redundant.

**Comment by danielmerino** [comment]
Good catch. I have removed the redundant property and I have bought "Boole algebra for dummies" book. ;-)

**Comment by juanjmerono** [comment]
You're going to kill me but I think you could use a predefined way to translate keys, for example using the prefix *moresite_* so you don't have to declare nothing but the **moresites.externalConfig.SiteTypes**, and add the keys with the prefix _moresite_xxxxx_ in the properties file. What do you think?

**Comment by danielmerino** [comment]
Hmm, I don't know. We save a property but IMHO we lose in configuration a bit of ease (people must notice that there is a mandatory hardcoded prefix "moresites_") and flexibility (maybe somebody for any reason wants to name these properties differently).

However, fixing this is a simple change. We can talk about it tomorrow in S2U call and if the consensus is saving properties, I'll do ASAP.

**Comment by juanjmerono** [comment]
Thanks Daniel !!

**Comment by danielmerino** [comment]
After a discussion in S2U meeting call, we decided to set two properties: a list of site types and a prefix to add to them to set up the properties names. I have pushed the latest changes, so you can remove the "in progress".

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java [hunk 2]
```diff
@@ -228,6 +228,21 @@ protected void processMySites()
 		// get Sections
 		Map<String, List> termsToSites = new HashMap<String, List>();
 		Map<String, List> tabsMoreTerms = new TreeMap<String, List>();
+		
+		//SAK-30712
+		String[] moresitesExternalSites = serverConfigurationService.getStrings("moresites.externalConfig.siteTypes");
+		String moresitesExternalPrefix = serverConfigurationService.getString("moresites.externalConfig.prefix");
```
Comment (juanjmerono) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java]: Please add "moresites_" as a default value here.

Comment (danielmerino) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/MoreSiteViewImpl.java]: Done.

**Pull request merged**